.github/workflows/platform-environment-matrix.yaml

```diff
@@ -26,12 +30,26 @@ jobs:
         name: Generate Matrix
         run: |
           # generate matrix
-          for f in environments/*/config.yaml ; do
-            if "${{ inputs.include_release }}"; then
-              [ -e "${f}" ] && e+="{\"environment\":\"$(yq .environment "${f}")\",\"platform\":\"$(yq .platform.vendor "${f}")\",\"release\":\"$(yq .release "${f}")\"},"
-            else
-              [ -e "${f}" ] && e+="{\"environment\":\"$(yq .environment "${f}")\",\"platform\":\"$(yq .platform.vendor "${f}")\"},"
-            fi
+          YAML_FILE=environments/environments.yaml 
+          include_all=${{ inputs.include_all }}
+
+          yaml_list={}
+          if [[ ! -f "$YAML_FILE" ]]; then
+            echo "File $YAML_FILE does not exist. Including all."
+            include_all=true
+          else
+              yaml_list=$(yq eval '.enabled[]' "$YAML_FILE")
+          fi
+
+          for f in environments/*/config.yaml; do
+              env=$(echo "$f" | cut -d'/' -f2)
```

Comment:
this makes the assumption that directory name matches the environment name, it's safer to extract it from the config.

also if no files exist it'll return null which might not do what you expect in the grep bellow, it's best to check the file exists first.

as that'd be the third `[ -e "${f}" ] &&` it might be worth wrapping the contents of the for loop in an ` if [ -e "${f}" ]` instead